MapBox.py

```diff
@@ -0,0 +1,226 @@
+import math
+import os
+import random
+import sys
+
+import threading
+import concurrent.futures
+import time
+
+from matplotlib import pyplot as plt
+import numpy as np
+
+import mapbox
+
+if getattr(sys, 'frozen', False):
+    local = os.path.dirname(sys.executable)
+elif __file__:
+    local = os.path.dirname(__file__)
+
+TILE_SIZE = 512
+
+
+def readKey():
+    if os.path.isfile(os.path.join(local, "apikey.txt")):
+        f = open(os.path.join(local, "apikey.txt"), "r")
+        contents = f.read()
+        return contents.strip()
+    else:
+        return None
+
+
+if not (readKey() is None):
+    maps = mapbox.Maps(access_token=readKey())
+else:
+    maps = None
+
+
+class MapPoint:
+
+    def __init__(self, latitude, longitude):
+        self.latitude = latitude
+        self.longitude = longitude
+        self.x = self.getX()
+        self.y = self.getY()
+
+    def getX(self):
+        return float(0.5 + self.longitude / 360)
+
+    def getY(self):
+        siny = math.sin(self.latitude * math.pi / 180)
+        return float((0.5 - math.log((1 + siny) / (1 - siny)) / (4 * math.pi)))
+
+    def getPixelX(self, zoom):
+        return int(math.floor(TILE_SIZE * (0.5 + self.longitude / 360) * math.pow(2, zoom)))
+
+    def getPixelY(self, zoom):
+        siny = math.sin(self.latitude * math.pi / 180)
+
+        # Truncating to 0.9999 effectively limits latitude to 89.189. This is
+        # about a third of a tile past the edge of the world tile.
+        siny = min(max(siny, -0.9999), 0.9999)
+
+        return int(
+            math.floor(TILE_SIZE * (0.5 - math.log((1 + siny) / (1 - siny)) / (4 * math.pi)) * math.pow(2, zoom)))
+
+    def getTileX(self, zoom):
+        return int(self.getPixelX(zoom) / TILE_SIZE)
+
+    def getTileY(self, zoom):
+        return int(self.getPixelY(zoom) / TILE_SIZE)
+
+
+## MapPoint Examples:
+p1 = MapPoint(49.284162, -123.2766960)  # NW of UBC
+p2 = MapPoint(49.239184, -123.210088)  # SE of UBC
+p3 = MapPoint(49.266063, -123.261348)  # NW of Vanier
+p4 = MapPoint(49.262715, -123.255472)  # SE of Vanier
+
+
+class MapTile:
+
+    def __init__(self, x, y, s):
+        self.x = x
+        self.y = y
+        self.s = s
+
+    def getName(self):
+        return f"{self.x}___{self.y}"
```

Comment:
I'm kinda nitpicking, but could we do with just one "_"?
```suggestion
        return f"{self.x}_{self.y}"
```